rita_common/src/rita_loop/write_to_disk.rs

```diff
@@ -0,0 +1,121 @@
+use crate::{
+    debt_keeper::send_debt_update, traffic_watcher::watch, tunnel_manager::tm_get_neighbors,
+};
+use std::{
+    thread,
+    time::{Duration, Instant},
+};
+
+use actix_async::System as AsyncSystem;
+use babel_monitor::{open_babel_stream, parse_routes};
+use settings::{
+    client::RitaClientSettings, exit::RitaExitSettingsStruct, get_rita_client, get_rita_exit,
+    FileWrite,
+};
+
+pub const FAST_LOOP_TIMEOUT: Duration = Duration::from_secs(4);
+// Save duration for all writes to disk in order to reduce write operations
+// pub const SAVING_TO_DISK_FREQUENCY: Duration = Duration::from_secs(600);
+#[derive(Clone)]
+pub enum OldSettingsType {
+    RitaClientSettings(RitaClientSettings),
+    RitaExitSettingsStruct(RitaExitSettingsStruct),
+}
+pub fn save_to_disk_loop(old_settings: OldSettingsType, file_path: &str, save_interval: u64) {
+    let file_path = file_path.to_string();
+
+    let saving_to_disk_frequency: Duration = Duration::from_secs(save_interval);
+    thread::spawn(move || loop {
+        let start = Instant::now();
+
+        if start.elapsed() < saving_to_disk_frequency {
+            thread::sleep(saving_to_disk_frequency - start.elapsed());
+        }
+
+        let router_mem_small;
+        //settings
+        match old_settings.clone() {
+            OldSettingsType::RitaClientSettings(old_settings_client) => {
+                let new_settings = get_rita_client();
+
+                if old_settings_client != new_settings {
+                    if let Err(e) = new_settings.write(&file_path) {
+                        warn!("writing updated config failed {:?}", e);
+                    }
+                    info!("Config/settings is being saved to");
+                }
+                match new_settings.network.device {
+                    Some(val) => router_mem_small = is_router_memory_small(&val),
+                    None => router_mem_small = false,
+                }
+            }
+            OldSettingsType::RitaExitSettingsStruct(old_settings_exit) => {
+                let new_settings = get_rita_exit();
+
+                if old_settings_exit != new_settings {
+                    if let Err(e) = new_settings.write(&file_path) {
+                        warn!("writing updated config failed {:?}", e);
+                    }
+                    info!("Config/settings is being saved to");
+                }
+                match new_settings.network.device {
+                    Some(val) => router_mem_small = is_router_memory_small(&val),
+                    None => router_mem_small = false,
+                }
+            }
+        }
+
+        // debt keeper
+        if !router_mem_small {
```

Comment:
this should document why we disable debt keeper on small storage devices, also should be router_storage_small not mem.